github/pkg/adapter/adapter_test.go

```diff
@@ -407,231 +409,108 @@ var testCases = []testCase{
-	}
-
-	h := &fakeHandler{
-		//handler: tc.sink,
-		handler: sinkAccepted, // No tests expect the sink to do anything interesting
+func (tc *testCase) validateAcceptedPayload(t *testing.T, ce *adaptertest.TestCloudEventsClient) error {
+	if len(ce.Sent()) != 1 {
+		return nil
 	}
-	sinkServer := httptest.NewServer(h)
-	defer sinkServer.Close()
 
-	ra, err := New(sinkServer.URL, testOwnerRepo)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	payload := gh.PullRequestPayload{}
-	subject, _ := strconv.ParseInt(testSubject, 10, 64)
-	payload.PullRequest.Number = subject
-	header := http.Header{}
-	header.Set("X-"+GHHeaderEvent, eventType)
-	header.Set("X-"+GHHeaderDelivery, eventID)
-	ra.HandleEvent(payload, http.Header(header))
-
-	// TODO(https://knative.dev/pkg/issues/250): clean this up when there is a shared test client.
-
-	canonicalizeHeaders(expectedRequest)
-	if diff := cmp.Diff(expectedRequest.Headers, h.header); diff != "" {
-		t.Errorf("Unexpected difference (-want, +got): %v", diff)
-	}
-
-	if diff := cmp.Diff(expectedRequest.Body, string(h.body)); diff != "" {
-		t.Errorf("Unexpected difference (-want, +got): %v", diff)
-	}
-}
-
-type fakeHandler struct {
-	body   []byte
-	header http.Header
-
-	handler func(http.ResponseWriter, *http.Request)
-}
+	got := ce.Sent()[0].Data()
 
-func (h *fakeHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	body, err := ioutil.ReadAll(r.Body)
+	data, err := json.Marshal(tc.payload)
 	if err != nil {
-		http.Error(w, "can not read body", http.StatusBadRequest)
-		return
+		return fmt.Errorf("Could not marshal sent payload: %v", err)
```

Comment:
Like I said in a previous comment, it would be better to fail here by calling `t.Fatal` instead of returning an error.

Reply:
Done.